test/rulesets/Base/mapreduce.jl

```diff
@@ -77,3 +77,41 @@
         end
     end # prod
 end
+
+@testset "Accumulations" begin
+    @testset "cumprod" begin
+        v = round.(10 .* randn(9), sigdigits=3)
+        test_rrule(cumprod, v)
+        v[3] = 0
+        test_rrule(cumprod, v)
+        v[6] = 0
+        test_rrule(cumprod, v)
+
+        @testset "higher dimensions, dims=$dims" for dims in (1,2,3)
+            m = round.(10 .* randn(4,5), sigdigits=3)
+            test_rrule(cumprod, m; fkwargs=(;dims=dims), atol=0.1)
+            m[2,2] = 0
+            m[2,4] = 0
+            test_rrule(cumprod, m; fkwargs=(;dims=dims))
+
+            t = round.(10 .* randn(3,3,3), sigdigits=3)
+            test_rrule(cumprod, t; fkwargs=(;dims=dims))
+            t[2,2,2] = 0
+            t[2,3,3] = 0
+            test_rrule(cumprod, t; fkwargs=(;dims=dims))
+        end
+
+        @testset "types" begin
+            back = unthunk(rrule(cumprod, [1, 2, 3])[2])  # allow integer input
+            @test unthunk(back(fill(0.5, 3))[2]) == [9/2, 2, 1]
```

Comment:
why are we testing values here?
It would be good to add comments explaining what we are particularly checking for that `test_rrule` will not catch

Reply:
`test_rrule` doesn't seem to accept integer input, this tests that the rule still does.
```
julia> test_rrule(cumprod, [1,2,3])
test_rrule: cumprod on Vector{Int64}: Error During Test at /Users/me/.julia/packages/ChainRulesTestUtils/6oOem/src/testers.jl:227
  Got exception outside of a @test
  InexactError: Int64(0.99)
```

Reply:
add comment to that effect?

Yeah can't test methods that require integers with finite differencing.
Since once you apply a finite difference you get a Float64 instead.
Which means you don't hit the method for integers.

Reply:
The comment now says "# rule allows integer input, but test_rrule does not"